drf_spectacular/contrib/rest_auth.py

```diff
@@ -0,0 +1,154 @@
+from django.conf import settings as global_settings
+
+from rest_framework import serializers
+
+from allauth.account import app_settings as allauth_settings
+
+from rest_auth.serializers import JWTSerializer, TokenSerializer
+
+from drf_spectacular.extensions import OpenApiViewExtension
+from drf_spectacular.utils import extend_schema
+
+SignupSerializer = global_settings.get('REST_AUTH_REGISTER_SERIALIZERS', {}).get('REGISTER_SERIALIZER', None)
+UserSerializer = global_settings.get('REST_AUTH_SERIALIZERS', {}).get('USER_DETAILS_SERIALIZER', None)
+
+if getattr(global_settings, 'REST_USE_JWT', False):
+    AuthTokenSerializer = JWTSerializer
+else:
+    AuthTokenSerializer = TokenSerializer
+
+
+class DRFDefaultDetailSerializer(serializers.Serializer):
+    detail = serializers.CharField(read_only=True, required=False)
+
+
+if allauth_settings.EMAIL_VERIFICATION == \
+                allauth_settings.EmailVerificationMethod.MANDATORY:
+    class SignupResponseSerializer(DRFDefaultDetailSerializer):
+        pass
+
+else:
+    class SignupResponseSerializer(serializers.Serializer):
+        user = UserSerializer(read_only=True)
+        token = AuthTokenSerializer(read_only=True)
+
+
+class RestAuthLoginView(OpenApiViewExtension):
+    target_class = 'rest_auth.views.LoginView'
+
+    def view_replacement(self):
+
+        class Fixed(self.target_class):
+
+            @extend_schema(
+                responses=AuthTokenSerializer,
+            )
+            def post(self, request, *args, **kwargs):
+                pass
+
+        return Fixed
+
+
+class RestAuthLogoutView(OpenApiViewExtension):
+    target_class = 'rest_auth.views.LogoutView'
+
+    def view_replacement(self):
+
+        class Fixed(self.target_class):
+
+            @extend_schema(
+                responses=
+                    (DRFDefaultDetailSerializer
+                     if getattr(global_settings, 'ACCOUNT_LOGOUT_ON_GET', None)
+                     else None),
```

Comment:
`None` needs to be http_method_not_allowed exception, so https://github.com/tfranzel/drf-spectacular/issues/101 is relevant here

Reply:
Actually it would be better to eliminate the `.get` altogether for this branch